collect.py

```diff
@@ -95,22 +95,37 @@ def get_all_attributes(driver, child):
     return child_attributes
 
 
+def get_elements_with_attributes(driver, elem_attributes, children):
+    elems_with_same_attributes = []
+    for child in children:
+        child_attributes = get_all_attributes(driver, child)
+        if child_attributes == elem_attributes:
+            elems_with_same_attributes.append(child)
+    return elems_with_same_attributes
+
+
 def do_something(driver, elem_attributes=None):
     elem = None
-    if elem_attributes is None:
-        body = driver.find_elements_by_tag_name('body')
-        assert len(body) == 1
-        body = body[0]
 
-        buttons = body.find_elements_by_tag_name('button')
-        links = body.find_elements_by_tag_name('a')
-        inputs = body.find_elements_by_tag_name('input')
-        selects = body.find_elements_by_tag_name('select')
-        children = buttons + links + inputs + selects
+    body = driver.find_elements_by_tag_name('body')
+    assert len(body) == 1
+    body = body[0]
+
+    buttons = body.find_elements_by_tag_name('button')
+    links = body.find_elements_by_tag_name('a')
+    inputs = body.find_elements_by_tag_name('input')
+    selects = body.find_elements_by_tag_name('select')
+    children = buttons + links + inputs + selects
+
+    if elem_attributes is None:
 
```

Comment:
Nit: sorry I missed this earlier, this newline should be removed here (there should be no newlines after `if`, `for`, etc.).